src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceDescription.cs

```diff
@@ -333,5 +351,20 @@ public ServiceDescription(IEnumerable<IServiceBehavior> injectedBehaviors, IServ
             AddBehaviors<TService>(this, behaviors);
             SetupSingleton<TService>(this, services);
         }
+
+        private static Lazy<Func<IServiceProvider, IEnumerable<IServiceBehavior>>> s_getKeyedServiceBehaviorsDelegate = new(() =>
+        {
+            Assembly assembly = Assembly.Load("Microsoft.Extensions.DependencyInjection.Abstractions");
+            Type type = assembly.GetType("Microsoft.Extensions.DependencyInjection.IKeyedServiceProvider");
+            ParameterExpression parameter = Expression.Parameter(typeof(IServiceProvider));
+            UnaryExpression cast = Expression.TypeAs(parameter, type);
```

Comment:
While the asp.net core built in service provider implements IKeyedServiceProvider, a 3rd party implementation might not. Need to have a check if the service provider implements it to avoid either a case exception or a null ref exception.

Reply:
good catch thanks!